### PR TITLE
 Switch saddr/daddr to u32's within tools/tcpconnlat.py 

### PR DESCRIPTION
**Pull request by kernel-z**
Switch saddr/daddr to u32's

### CODE REVIEW

File: tools/tcpconnlat.py [hunk 1]
```diff
@@ -84,8 +84,8 @@ def positive_float(val):
     // XXX: switch some to u32's when supported
     u64 ts_us;
     u64 pid;
-    u64 saddr;
-    u64 daddr;
+    u32 saddr;
+    u32 daddr;
     u64 ip;
     u64 dport;
```
Comment (yonghong-song) [inline tools/tcpconnlat.py]: could you help convert pid to u32 and dport to u16 as well? Once this is done, you can remove the comments.

The same for below ipv6_data_t data structure.

File: tools/tcpconnlat.py [hunk 2]
```diff
@@ -206,23 +206,23 @@ def positive_float(val):
 class Data_ipv4(ct.Structure):
     _fields_ = [
         ("ts_us", ct.c_ulonglong),
-        ("pid", ct.c_ulonglong),
+        ("pid", ct.c_uint),
         ("saddr", ct.c_uint),
         ("daddr", ct.c_uint),
         ("ip", ct.c_ulonglong),
-        ("dport", ct.c_ulonglong),
+        ("dport", ct.c_uint),
```
Comment (yonghong-song) [inline tools/tcpconnlat.py]: dport should be ct.c_ushort?

Comment (kernel-z) [reply tools/tcpconnlat.py]: yeah,ct.c_uint also works,  had changed to ct.c_ushort. Thanks

**Comment by brendangregg** [comment]
Thanks, please make sure these changes are tested, on both IPv4 and IPv6. I'm not 100% certain we can fix all these, but it's worth trying.

**Comment by kernel-z** [comment]
Hi,@brendangregg, I had testted with IPv4 and IPv6.

> PID    COMM         IP SADDR            DADDR            DPORT LAT(ms)
2583   tcpcli01     4  192.168.99.101   192.168.99.100   9877  0.40
2584   tcpcli01     4  192.168.99.101   192.168.99.100   9877  0.39
2585   tcpcli01     4  192.168.99.101   192.168.99.100   9877  1.24
2586   tcpcli01     4  192.168.99.101   192.168.99.100   9877  0.42
2587   tcpcli01     4  192.168.99.101   192.168.99.100   9877  2.21
2588   client6      6  2400:a480:aaaa:200::159 2400:a480:aaaa:200::158 9877  0.59
2589   client6      6  2400:a480:aaaa:200::159 2400:a480:aaaa:200::158 9877  0.35
2590   client6      6  2400:a480:aaaa:200::159 2400:a480:aaaa:200::158 9877  0.47

I think it works.

**Comment by pchaigno** [comment]
I tested the changes as well, on Linux 4.4, 4.9, and 4.17, with LLVM 5.0.2. I opened a connection on the loopback interface with both IPv4 and IPv6 each time. LGTM!

**Comment by yonghong-song** [comment]
[buildbot, ok to test]

**Comment by yonghong-song** [comment]
LGTM. Thanks everyone!

**Pull request merged**